backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/tutor/question/QuestionService.java

```diff
@@ -85,7 +101,7 @@ public QuestionDto findQuestionByKey(Integer key) {
       backoff = @Backoff(delay = 5000))
     @Transactional(isolation = Isolation.READ_COMMITTED)
     public List<QuestionDto> findQuestions(int courseId) {
-        return questionRepository.findQuestions(courseId).stream().map(QuestionDto::new).collect(Collectors.toList());
+        return questionRepository.findQuestions(courseId).stream().filter(q -> !q.isSubmittedQuestion()).map(QuestionDto::new).collect(Collectors.toList());
```

Comment:
due to the number of invocations here it make sense previous comment about the performance of isSubmittedQuestion